specs/Qowaiv.CodeAnalysis.Specs/Cases/DefinePropertiesAsImmutables.cs

```diff
@@ -0,0 +1,147 @@
+
+        public static int StaticProperties { get; set; } // Compliant {{Static properties are ignored.}}
+
+        public int PublicField; // Compliant {{Fields are ignored.}}
+
+        public int InitOnly { get; init; } // Compliant {{Property is can only be changed during the init phase.}}
+
+        public int Caclulated => 42 * PublicField; // Compliant {{Calculated properties are considered immutable.}}
+
+        public int GetOnly { get; } // Compliant {{Property is set during construction.}}
+
+        internal int Internal { get; set; } // Compliant {{Internal properties are ignored.}}
+
+        internal protected int InternalProtected { get; set; } // Compliant {{Internal protected properties are ignored.}}
+
+        int WithoutAccessors { get; set; } // Compliant {{Properties without accessors are private and hence ignored.}}
+
+        private int Private { get; set; } // Compliant {{Private properties are ignored.}}
+    }
+
+    public record Record
+    {
+        public Record(int val) => GetOnly = val;
+
+        public int GetOnly { get; } // Compliant {{Property is set during construction.}}
+
+        public int InitOnly { get; init; } // Compliant {{Property is can only be changed during the init phase.}}
+    }
+
+    public struct Struct
+    {
+        public Struct(int val) => GetOnly = val;
+
+        public int GetOnly { get; } // Compliant {{Property is set during construction.}}
+    }
+
+    public ref struct RefStruct
+    {
+        public int Value { get; set; } // Compliant {{Ref structs are intended mutable.}}
+    }
+
+    [Mutable]
+    public class DecoratedClass
+    {
+        public int Value { get; set; } // Compliant {{Class is decorated.}}
+    }
+
+    [Configuration]
+    public class DecoratedWithDerivedClass
+    {
+        public int Value { get; set; } // Compliant {{Class is decorated.}}
+    }
+
+    [Obsolete]
+    public class ObsoleteClass
+    {
+        public int Value { get; set; } // Compliant {{Class is obsolete.}}
+    }
+
+    public class ObsoleteProperty
+    {
+        [Obsolete]
+        public int Value { get; set; } // Compliant {{Class is obsolete.}}
+    }
+}
+
+namespace Noncompliant
+{
+    public class Class
+    {
+        public int PublicSet { get; set; } // Noncompliant {{Remove this setter, or make this property init-only.}}
+        //                          ^^^^
+        
+        public int ProtectedSet { get; protected set; } // Noncompliant 
+        //                             ^^^^^^^^^^^^^^
+
+        public int PrivateSet { get; private set; } // Noncompliant
+
+        protected int Protected { get; set; } // Noncompliant
+        
+        protected int ProtectedPrivateSet { get; private set; } // Noncompliant
+
+        public int WithUnderlyingField
+        {
+            get => underlying;
+            set => underlying = value; // Noncompliant
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^
+        }
+
+        public int WithUnderlyingFieldClasci
```

Comment:
Clasci doesn't make sense to me - I get what you're testing here but maybe rename the object

Reply:
Classic... ;)